### PR TITLE
Adds Simulated Annealing diagram

### PR DESCRIPTION
**Pull request by Rishav159**
-This removes the unnecessary code for applying fill from the Hill Climbing Search Diagram
-Adds visualization for Simulated Annealing. The animation has a slider with decreasing temperature automatically. The reader can also observe the behavior of the algorithm at a particular temperature by holding on to that value on the slider.

### CODE REVIEW

File: 4-Beyond-Classical-Search/c_simulatedAnnealing.js [hunk 1]
```diff
@@ -1,82 +1,152 @@
...
+    destroy() {
+      this.stopAnnealing();
+      this.pauseAnimation();
+    }
 
-	function setupScene(){
-		// If background already exists,
-		// then clear it
-		if(background != null)
-		two.remove(background);
```
Comment (redblobgames) [inline 4-Beyond-Classical-Search/c_simulatedAnnealing.js]: Are you still using Two.js here?

**Comment by Rishav159** [comment]
No I am not. The deleted part is from the past implementation.

File: 4-Beyond-Classical-Search/c_simulatedAnnealing.js (hunk 1 above)
Comment (Rishav159) [inline 4-Beyond-Classical-Search/c_simulatedAnnealing.js]: No I am not. The deleted part is from the past implementation.

Comment (redblobgames) [inline 4-Beyond-Classical-Search/c_simulatedAnnealing.js]: Oh, silly me, I misread the diff :)

**Pull request merged**